packages/transport/src/transports/abstractUsb.ts

```diff
@@ -85,56 +85,55 @@ export abstract class AbstractUsbTransport extends AbstractTransport {
     }
 
     public acquire({ input }: { input: AcquireInput }) {
-        return this.scheduleAction(async () => {
-            const { path } = input;
+        return this.scheduleAction(
+            async () => {
+                const { path } = input;
+
+                if (this.listening) {
+                    this.listenPromise[path] = createDeferred<string>();
+                    this.listenPromise[path].promise
+                        .catch(err => err)
+                        .finally(() => {
+                            delete this.listenPromise[path];
```

Comment:
Are we sure that `this.listenPromise[path]` at this time is the same as when we assigned it four lines above? Or could it be a different one e.g. if we use the class incorrectly?